lib/Onyx.js

```diff
@@ -150,6 +150,45 @@ function addAllSafeEvictionKeysToRecentlyAccessedList() {
         });
 }
 
+/**
+ * When a collection of keys change, search for any callbacks matching the collection key and trigger those callbacks
+ *
+ * @param {String} parentKey
+ * @param {Object} collection
+ */
+function keysChanged(parentKey, collection) {
+    // Add or remove this parentKey from the recentlyAccessedKeys lists
+    if (!_.isNull(collection)) {
+        addLastAccessedKey(parentKey);
+    } else {
+        removeLastAccessedKey(parentKey);
+    }
+
+    // Find all subscribers that were added with connect() and trigger the callback or setState() with the new data
```

Comment:
>trigger the callback

We are not triggering any callback here only `setState()`. However, we probably need to?

Each time we use `Onyx.connect()` with a collection key like `ONYXKEYS.COLLECTION.REPORTS` that callback will expect to see receive an update for each key updated.

It makes sense that we might not want this to happen for a React component since it would re-render n report count times which is bad for performance.

But for the subscribers not using `withOnyx()` I think we should still send updates to them? Otherwise a call to `mergeCollection()` will fail to update it's non React subscribers.

Reply:
e.g. the `handleReportChanged()` callback here will now not be updated when we merge a collection

https://github.com/Expensify/Expensify.cash/blob/adfdb884d1ccc726893c73dc1d331631b477dd42/src/libs/actions/Report.js#L645-L663

Reply:
We are triggering a setState. We're not doing a callback though.

Ah you're right about the handleReportChanged. I was just testing places that connected to the collection via withOnyx. Yeah i think with the current code it does not send updates to connections via onyx. Yup, i think we should send updates to them as well. I'll update and test it out. I might have to change their format from report object to reports array but I'll check it out.

Reply:
Updated.